chain/epoch_manager/src/lib.rs

```diff
@@ -929,34 +915,83 @@ impl EpochManager {
                 validator_to_shard[*validator_id as usize].insert(shard_id as ShardId);
             }
         }
-        let epoch_info_aggregator =
-            self.get_and_update_epoch_info_aggregator(&epoch_id, block_hash, true)?;
-        let current_validators = cur_epoch_info
-            .validators
-            .into_iter()
-            .enumerate()
-            .map(|(validator_id, info)| {
-                let validator_stats = epoch_info_aggregator
-                    .block_tracker
-                    .get(&(validator_id as u64))
-                    .unwrap_or_else(|| &ValidatorStats { produced: 0, expected: 0 })
-                    .clone();
-                let mut shards =
-                    validator_to_shard[validator_id].clone().into_iter().collect::<Vec<ShardId>>();
-                shards.sort();
-                Ok(CurrentEpochValidatorInfo {
-                    is_slashed: slashed.contains_key(&info.account_id),
-                    account_id: info.account_id,
-                    public_key: info.public_key,
-                    stake: info.stake,
-                    shards,
-                    num_produced_blocks: validator_stats.produced,
-                    num_expected_blocks: validator_stats.expected,
-                })
-            })
-            .collect::<Result<Vec<CurrentEpochValidatorInfo>, EpochError>>()?;
+
+        // This ugly code arises because of the incompatible types between `block_tracker` in `EpochInfoAggregator`
+        // and `validator_block_chunk_stats` in `EpochSummary`. Rust currently has no support for Either type
+        // in std.
```

Comment:
we have `either` crate in crate graph. But also, urgh, I dont' know how to make this meaningfully better :man_shrugging:

Reply:
Thought about it but decide not to introduce unnecessary dependencies :)